idaes/models_extra/power_generation/unit_models/soc_submodels/channel.py

```diff
@@ -0,0 +1,839 @@
+# Framework (IDAES IP) was produced under the DOE Institute for the
+# Design of Advanced Energy Systems (IDAES), and is copyright (c) 2018-2021
+# by the software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia University
+# Research Corporation, et al.  All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and
+# license information.
+#################################################################################
+
+__author__ = "John Eslick, Douglas Allan"
+
+from pyomo.common.config import ConfigValue, In
+from pyomo.dae import DerivativeVar
+import pyomo.environ as pyo
+from pyomo.network import Port
+
+
+from idaes.core import declare_process_block_class, UnitModelBlockData, useDefault
+import idaes.models_extra.power_generation.unit_models.soc_submodels.common as common
+from idaes.models_extra.power_generation.unit_models.soc_submodels.common import (
+    _constR,
+    _set_if_unfixed,
+    _species_list,
+    _element_list,
+    _element_dict,
+)
+import idaes.core.util.scaling as iscale
+from idaes.core.solvers import get_solver
+
+from idaes.core.base.var_like_expression import VarLikeExpression
+import idaes.logger as idaeslog
+
+
+@declare_process_block_class("SocChannel")
+class SocChannelData(UnitModelBlockData):
+    CONFIG = UnitModelBlockData.CONFIG()
+    CONFIG.declare(
+        "component_list",
+        ConfigValue(default=["H2", "H2O"], description="List of components"),
+    )
+
+    CONFIG.declare(
+        "control_volume_zfaces",
+        ConfigValue(
+            description="List containing coordinates of control volume faces "
+            "in z direction. Coordinates must start with zero, be strictly "
+            "increasing, and end with one"
+        ),
+    )
+    CONFIG.declare(
+        "length_z",
+        ConfigValue(
+            default=None, description="Length in the direction of flow (z-direction)"
+        ),
+    )
+    CONFIG.declare(
+        "length_y", ConfigValue(default=None, description="Width of cell (y-direction)")
+    )
+    CONFIG.declare(
+        "include_temperature_x_thermo",
+        ConfigValue(
+            domain=In([useDefault, True, False]),
+            default=True,
+            description="Whether to consider temperature variations in "
+            "x direction in thermodynamic equations",
+        ),
+    )
+    common._thermal_boundary_conditions_config(CONFIG, thin=False)
+    CONFIG.declare(
+        "opposite_flow",
+        ConfigValue(default=False, description="If True assume velocity is negative"),
+    )
+    CONFIG.declare(
+        "below_electrode",
+        ConfigValue(
+            domain=In([True, False]),
+            description="Decides whether or not to create material "
+            "flux terms above or below the channel.",
```

Comment:
Maybe for description "Set the electrode boundary (material flux) location." There is a minor disconnect between the description and the option name since the description doesn't mention the electrode.

Reply:
Further - the options here are "True" and "False" but the text (to a naïve reader at least) implies it should be "Above" or "below".